shell/common/shell.h

```diff
@@ -101,7 +99,9 @@ class Shell {
   TracingController tracing_controller_;
 
   std::vector<ftl::WeakPtr<Rasterizer>> rasterizers_;
-  std::vector<ftl::WeakPtr<PlatformView>> platform_views_;
+  std::vector<std::weak_ptr<PlatformView>> platform_views_;
+
+  std::recursive_mutex mutex_;
```

Comment:
Ah, now i see that it's also being used to guard the rasterizers list.  I'd suggest using two separate locks for platform_views and rasterizers.